### PR TITLE
fix getUniform API bug

### PR DESCRIPTION
**Pull request by moshuying**
Re: #

### Changelog

* sometime get uniform will be crash in native scene

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/math/Color.cpp [hunk 1]
```diff
@@ -93,6 +97,13 @@ void Color::set(const Color &p1, const Color &p2) {
     a = p2.a - p1.a;
 }
 
+void Color::set(const Vec4 &v) {
+    r = v.x * 255.F;
+    g = v.y * 255.F;
+    b = v.z * 255.F;
+    a = v.w * 255.F;
+}
```
Comment (dumganhar) [inline native/cocos/math/Color.cpp]: Revert the changes in Color.h/.cpp, they are not needed now.

**Comment by dumganhar** [comment]
Test code in HelloWorld project:
```ts
import { Vec4 } from 'cc';
import { _decorator, Color, Component, gfx, native, Node, SkinnedMeshRenderer, Vec3 } from 'cc';
import { NATIVE } from 'cc/env';
const { ccclass, property } = _decorator;

@ccclass('MyTest')
export class MyTest extends Component {
    start() {
        const renderer = this.getComponent(SkinnedMeshRenderer);
        const mat = renderer.materials[0];
        const pass = mat.passes[1];
        
        {
            const handle = pass.getHandle('mainColor', 0, gfx.Type.FLOAT4);
            let uniformValue = new Color();
            uniformValue = pass.getUniform(handle, uniformValue);
            console.log(`cjh mainColor: ${uniformValue}`);
        }
        
        {
            let colorScaleHandle = pass.getHandle('colorScale');
            let value = new Vec3();
            console.log(`cjh init colorScale: ${value}`);
            value = pass.getUniform(colorScaleHandle, value);
            console.log(`cjh colorScale: ${value}`);
        }

        {
            let handle = pass.getHandle('baseStep');
            let value = 111;
            value = pass.getUniform(handle, value);
            console.log(`cjh baseStep: ${value}`);
        }

        {
            let handle = pass.getHandle('shadeFeather');
            let value = 111;
            value = pass.getUniform(handle, value);
            console.log(`cjh shadeFeather: ${value}`);
        }

        {
            let prop = mat.getProperty('mainTexture', 1);
            console.log(`cjh mainTexture: ${prop}`);
        }

        {
            let handle = pass.getHandle('tilingOffset');
            let value = new Vec4();
            value = pass.getUniform(handle, value);
            console.log(`cjh tilingOffset: ${value}`);
        }

        setInterval(function(){
            if (NATIVE) {
                native.garbageCollect();
            }
        }, 1000);
    }
}
```

**Review by dumganhar — APPROVED** [review]

**Review by PatriceJiang — APPROVED** [review]

File: native/cocos/math/Color.cpp [hunk 2]
```diff
@@ -50,6 +50,10 @@ Color::Color(uint32_t val) {
     set(val);
 }
 
+//Color::Color(const Vec4 &v) {
+//    set(v);
+//}
+
```
Comment (minggo) [inline native/cocos/math/Color.cpp]: Please remove unneeded codes.

File: native/cocos/math/Color.h [hunk 3]
```diff
@@ -61,6 +61,8 @@ class Color {
     explicit Color(const uint8_t *src);
 
     explicit Color(uint32_t val);
+    
+ //   explicit Color(const Vec4 &v);
```
Comment (minggo) [inline native/cocos/math/Color.h]: Why remove it?

File: native/cocos/math/Color.cpp (hunk 1 above)
Comment (moshuying) [reply native/cocos/math/Color.cpp]: ok, this change reverted.

File: native/cocos/math/Color.cpp (hunk 2 above)
Comment (moshuying) [reply native/cocos/math/Color.cpp]: ok, unused comment has been removed

File: native/cocos/math/Color.h (hunk 3 above)
Comment (moshuying) [reply native/cocos/math/Color.h]: No need to remove, This change reverted

**Review by minggo — APPROVED** [review]
